### ISSUE TITLE
[google_maps_flutter] [ios] Unpin GoogleMaps cocoapod.

### ISSUE THREAD

**Issue by ditman** [flutter/flutter#63530]
There seems to be a problem with the GoogleMaps 3.10 cocoapod, requiring XCode 11.4 to compile in an undocumented way.

Our test runners currently use XCode 11.3.1, so when the google_maps_flutter plugin started getting the GoogleMaps 3.10 pod, builds started to fail.

The issue was investigated and mitigated [here](https://github.com/flutter/plugins/pull/2924).

**Remove the pin to `'<3.9'` on google_map_flutter's podspec file** once:

A) We decide XCode 11.4 is required to build flutter Apps, and our infra has been updated.
B) GoogleMaps (pod) behaves nicely again.

There are two internal issues for GoogleMaps 3.10 related to this issue that can be monitored to see the progress of "B":

* b/163359804
* b/163474612

**Comment by jmagman** [comment]
Even if they fix in 3.10.1, you will probably never be able to allow 3.10.

**Comment by ditman** [comment]
I'd expect that once they publish 3.10.1 cocoapod will never resolve to the older 3.10.0?

**Comment by jmagman** [comment]
🤷‍♀️  Depends on if other dependencies use `GoogleMaps` and their defined ranges.  Ideally we should be explicit.

**Comment by ditman** [comment]
> A list of version requirements can be specified for even more fine grained control.

https://guides.cocoapods.org/syntax/podfile.html#pod

I guess it can be something like: `'< 3.10', '> 3.10'` to "remove" the pin? (while excluding `3.10`)

**Comment by hectorAguero** [comment]
So, at the time GoogleMaps iOS SDK is in version 4.2.0 and the flutter_package still use the 3.9.0 SDK for iOS, also they announce drop ios 10 support for version 5.0, can the google maps flutter package use the latest iOS SDK?

https://developers.google.com/maps/documentation/ios-sdk/releases

**Comment by jmagman** [comment]
Flutter 2 still has a minimum of Xcode 11.0, but recommends 12.0.1.  In any case, the infra story is past Xcode 12 now.
Maybe the best thing is to totally unpin it, and then users can set `pod 'GoogleMaps', '< 3.10'` or whatever versions in their own Podfile depending on what Xcode they have installed, their minimum target iOS version, etc...
What do you think, @ditman?

**Comment by ditman** [comment]
@jmagman yep, I think it's worth a try. I created this issue while gardening and iOS is not my jam, but this looks safe.

After some time, the 3.10 dependency on a more modern XCode was documented, btw!

```
Note:

To build apps using the Maps SDK for iOS, the current minimum IDE version supported is Xcode 11.5.
```

If we're past that, we should be fine, I guess?

**Comment by jmagman** [comment]
> If we're past that, we should be fine, I guess?

Our infra is past that, but the minimum required on Flutter 2 isn't.  I think unpinning is probably the best idea since individual users can have control over what `GoogleMaps` version they use.  But if the plugin doesn't unpin, they can't upgrade past 3.9 even if the have the right SDK version installed.
